src/engraving/dom/edit.cpp

```diff
@@ -4069,83 +4069,107 @@ MeasureBase* Score::insertMeasure(ElementType type, MeasureBase* beforeMeasure,
             std::vector<Clef*> clefList;
             std::vector<Clef*> previousClefList;
             std::list<Clef*> specialCaseClefs;
+            std::vector<BarLine*> previousBarLinesList;
 
             //
-            // remove clef, time and key signatures
+            // remove clef, barlines, time and key signatures
             //
-            if (options.moveSignaturesClef && measureInsert) {
+            if (measureInsert) {
                 for (size_t staffIdx = 0; staffIdx < score->nstaves(); ++staffIdx) {
-                    Measure* pm = measureInsert->prevMeasure();
+                    Measure* pm = newMeasure->prevMeasure();
                     if (pm) {
                         Segment* ps = pm->findSegment(SegmentType::Clef, tick);
                         if (ps && ps->enabled()) {
                             EngravingItem* pc = ps->element(staffIdx * VOICES);
-                            if (pc) {
+                            if (pc && !options.moveSignaturesClef) {
```

Comment:
It looks like we can move the `!options.moveSignaturesClef` a few `if` conditions up, to save some function calls.